aws-lc-rs/src/aead.rs

```diff
@@ -1237,20 +1205,18 @@ impl TlsRecordSealingKey {
 /// AEAD Encryption key used for TLS protocol record encryption.
 ///
 /// This type encapsulates decryption operations for TLS AEAD algorithms.
-/// It validates that the provided nonce values are monotonically increasing for each invocation.
 ///
 /// The following algorithms are supported:
 /// * `AES_128_GCM`
 /// * `AES_256_GCM`
 ///
-/// # FIPS
 /// Use this type in place of `LessSafeKey`, `OpeningKey`, `SealingKey`.
 pub struct TlsRecordOpeningKey {
     // The TLS specific construction for TLS ciphers in AWS-LC are not thread-safe!
     // The choice here was either wrap the underlying EVP_AEAD_CTX in a Mutex as done here,
     // or force this type to !Sync. Since this is an implementation detail of AWS-LC
     // we have optex to manage this behavior internally.
-    ctx: Mutex<AeadCtx>,
+    ctx: AeadCtx,
```

Comment:
Was the `Mutex` needed here? If so, we should also change the "open" methods to take an exclusive/mut reference.

Reply:
As discussed in the meeting the underlying AeadCtx on the open path doesn't have any mutable state. So the call out was that it neither needs the mutex nor does it need the &mut self.